cypress/integration/apprenticeFilterSpec.js

```diff
@@ -0,0 +1,75 @@
+/// <reference types="cypress" />
+
+describe('Should filter Apprenticeships correctly', () => {
+  beforeEach(() => {
+    cy.visit('/')
+  })
+
+  it('filters all', function() {
+    cy.get(`#filters-all`).click();  
+    cy.get('.list__item__content-category').contains('software')
+  });
+
+  const filters = ['development', 'design', 'software', 'mobile']
+  filters.map(filter => {
+    return it(`filters ${filter}`, function() {
+      cy.get(`#filters-${filter}`).click()
+      cy.get('.list__item__content-category').contains(`${filter}`)
+    });
+  })
+})
+
+
+let Apprenticeship;
+Apprenticeship = (() => {
+  function Apprenticeship(name, category, hidden) {
+    this.name = name;
+    this.category = category;
+    this.hidden = hidden != null ? hidden : false;
+  }
+
+  Apprenticeship.prototype.hide = function() {
+    return this.hidden = true;
+  };
+
+  Apprenticeship.prototype.show = function() {
+    return this.hidden = false;
+  };
+
+  return Apprenticeship;
+})();
+
+describe('Should hide/show Apprenticeship & have category, hidden, & name properties', () => {
```

Comment:
Sorry for the confusion, we don't need to move these unit tests over to Cypress. On closer inspection, I don't think the Jasmine tests were actually testing real production code anyway.  They're testing the `Apprenticeship` object defined within the test file, not the one that is actually used in the application (`source/js/app.js`). It looks to me like this may have just been an abandoned exercise in unit testing. 

I don't think it's worth going back to add unit tests around `Apprenticeship` in `app.js` at this point, because your other Cypress tests will cover the same thing that these tests are exercising, that an apprenticeship can be displayed or hidden, just at a higher level. 

In a larger or more complex web application, I'd advocate for robust unit tests along with E2E tests. But I think this site is small enough that the new filtering tests you added will be sufficient, and we can just remove these.

Reply:
Yes I agree